### PR TITLE
[hypo71] Fix arrival count measures

### PR DESCRIPTION
**Pull request by gempa-dirk**

### CODE REVIEW

**Comment by gempa-dirk** [comment]
@jmsaurel Please comment the PR and allow us to merge it into master.

**Comment by jmsaurel** [comment]
@gempa-dirk this PR looks ok for me.
I do not see anything obviously wrong or erroneous.

File: plugins/locator/hypo71/hypo71.cpp [hunk 1]
```diff
@@ -2231,25 +2231,25 @@ Origin* Hypo71::locate(PickList& pickList) {
 						origin->removeArrival(sArrival->index());
 						origin->add(sArrival.get());
 					}
+					if ( sArrival->timeUsed() ) {
```
Comment (gempa-jabe) [inline plugins/locator/hypo71/hypo71.cpp]: This can throw an exception.

File: plugins/locator/hypo71/hypo71.cpp [hunk 2]
```diff
@@ -2205,9 +2201,13 @@ Origin* Hypo71::locate(PickList& pickList) {
 					pArrival->setTakeOffAngle(toDouble(takeOffAngle));
 
 					if ( !origin->add(pArrival.get()) ) {
+						SEISCOMP_DEBUG("remove");
 						origin->removeArrival(pArrival->index());
 						origin->add(pArrival.get());
 					}
+					if ( pArrival->timeUsed() ) {
```
Comment (gempa-jabe) [inline plugins/locator/hypo71/hypo71.cpp]: This can throw an exception.

**Review by gempa-jabe — CHANGES_REQUESTED** [review]

File: plugins/locator/hypo71/hypo71.cpp (hunk 1 above)
Comment (gempa-dirk) [reply plugins/locator/hypo71/hypo71.cpp]: try ... catch added

File: plugins/locator/hypo71/hypo71.cpp (hunk 2 above)
Comment (gempa-dirk) [reply plugins/locator/hypo71/hypo71.cpp]: try ... catch added

File: plugins/locator/hypo71/hypo71.cpp [hunk 3]
```diff
@@ -2205,9 +2201,17 @@ Origin* Hypo71::locate(PickList& pickList) {
...
 					if ( !origin->add(pArrival.get()) ) {
+						SEISCOMP_DEBUG("remove");
 						origin->removeArrival(pArrival->index());
 						origin->add(pArrival.get());
 					}
+					try {
+						if ( pArrival->timeUsed() ) {
+							usedAssocCount++;
+						}
+					} catch ( exception& e ) {
```
Comment (gempa-jabe) [inline plugins/locator/hypo71/hypo71.cpp]: Newline before `catch`.

File: plugins/locator/hypo71/hypo71.cpp [hunk 4]
```diff
@@ -2231,25 +2235,29 @@ Origin* Hypo71::locate(PickList& pickList) {
 						origin->removeArrival(sArrival->index());
 						origin->add(sArrival.get());
 					}
+					try {
+						if ( sArrival->timeUsed() ) {
+							++usedAssocCount;
+						}
+					} catch ( exception& e ) {
```
Comment (gempa-jabe) [inline plugins/locator/hypo71/hypo71.cpp]: Newline

**Review by gempa-jabe — CHANGES_REQUESTED** [review]

File: plugins/locator/hypo71/hypo71.cpp (hunk 3 above)
Comment (gempa-dirk) [reply plugins/locator/hypo71/hypo71.cpp]: adjusted

**Review by gempa-jabe — APPROVED** [review]

**Pull request merged**